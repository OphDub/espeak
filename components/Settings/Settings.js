import React, { Component } from 'react';

import {
  Text,
  View,
  StyleSheet,
  TouchableOpacity
} from 'react-native';

type Props = {};

export default class Home extends Component <Props> {
  render() {
    const { name, points, stack_id } = this.props.screenProps.user;
    const { handleSignOut } = this.props.screenProps;
    return (
      <View style={styles.container}>
        <View style={styles.statsDisplay}>
          <Text style={styles.listItems}>Username: {name}</Text>
          <Text style={styles.listItems}>Current Points: {points}</Text>
          <Text style={styles.listItems}>Current Level: {stack_id}</Text>
        </View>
        <TouchableOpacity
          onPress={() => handleSignOut()}
          style={styles.logoutBtn}
        >
          <Text style={styles.buttonText}>LOGOUT</Text>
        </TouchableOpacity>
      </View>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'column',
    justifyContent: 'center',
    paddingVertical: 50,
  },
  statsDisplay: {
    display: 'flex',
    alignSelf: 'center',
    width: 280,
    paddingHorizontal: 20,
    paddingVertical: 30,
    backgroundColor: 'white',
    borderColor: '#1e3888',
    borderWidth: 2,
    borderRadius: 8,
    marginVertical: 20,
    shadowColor: '#979797',
    shadowOpacity: 0.7,
    shadowOffset: { width: 2, height: 2 },
    shadowRadius: 4,
  },
  listItems: {
    fontSize: 24,
    margin: 5,
    color: '#1e3888'
  },
  logoutBtn: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
    alignSelf: 'center',
    margin: 10,
    padding: 10,
    width: 150,
    backgroundColor: '#3AAFB9',
    borderRadius: 8,
    shadowColor: '#979797',
    shadowOpacity: 0.7,
    shadowOffset: { width: 2, height: 2 },
    shadowRadius: 4,
  },
  buttonText: {
    color: '#FFF',
    fontSize: 18,
  }
});
