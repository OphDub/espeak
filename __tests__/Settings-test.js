import React from 'react';
import { shallow, configure } from 'enzyme';
import Settings from '../components/Settings/Settings';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
configure({ adapter: new Adapter() });

it('renders correctly', () => {
  const wrap = renderer.create(<Settings screenProps={{userEmail: 'a@z.com'}}/>).toJSON();
  expect(wrap).toMatchSnapshot();
})