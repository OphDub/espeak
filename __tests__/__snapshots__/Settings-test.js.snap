// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "paddingVertical": 50,
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "white",
        "borderColor": "#1e3888",
        "borderRadius": 8,
        "borderWidth": 2,
        "display": "flex",
        "marginVertical": 20,
        "paddingHorizontal": 20,
        "paddingVertical": 30,
        "shadowColor": "#979797",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.7,
        "shadowRadius": 4,
        "width": 280,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1e3888",
          "fontSize": 24,
          "margin": 5,
        }
      }
    >
      Username: 
      a@z.com
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1e3888",
          "fontSize": 24,
          "margin": 5,
        }
      }
    >
      Current Points: 
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#1e3888",
          "fontSize": 24,
          "margin": 5,
        }
      }
    >
      Current Level:
    </Text>
  </View>
  <View
    accessibilityComponentType={undefined}
    accessibilityLabel={undefined}
    accessibilityTraits={undefined}
    accessible={true}
    collapsable={undefined}
    hasTVPreferredFocus={undefined}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "center",
        "backgroundColor": "#3AAFB9",
        "borderRadius": 8,
        "display": "flex",
        "justifyContent": "center",
        "margin": 10,
        "opacity": 1,
        "padding": 10,
        "shadowColor": "#979797",
        "shadowOffset": Object {
          "height": 2,
          "width": 2,
        },
        "shadowOpacity": 0.7,
        "shadowRadius": 4,
        "width": 150,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "#FFF",
          "fontSize": 18,
        }
      }
    >
      LOGOUT
    </Text>
  </View>
</View>
`;
